httpclient5/src/main/java/org/apache/hc/client5/http/DnsResolver.java

```diff
@@ -61,4 +67,21 @@ public interface DnsResolver {
      */
     String resolveCanonicalHostname(String host) throws UnknownHostException;
 
+    /**
+     * Returns a list of {@link SocketAddress} for the given host with the given port.
+     *
+     * @see SocketAddress
+     *
+     * @since 5.5
+     */
+    default List<SocketAddress> resolve(String host, int port) throws UnknownHostException {
```

Comment:
@arturobernalg Almost there. Let's change this method to return `List<InetSocketAddress>` in order to avoid ugly downcasts in several places.

Reply:
done